src/de.f90

```diff
@@ -23,18 +23,55 @@ module de
 
 
   !Main differential evolution routine.
-  subroutine diver(func, lowerbounds, upperbounds, path, nDerived, discrete, partitionDiscrete,            &
-                   maxciv, maxgen, NP, F, Cr, lambda, current, expon, bndry, jDE, lambdajDE,               &
-                   convthresh, convsteps, removeDuplicates, doBayesian, prior, maxNodePop, Ztolerance,     &
-                   savecount, resume, outputSamples, init_population_strategy, discard_unfit_points,       &
-                   max_initialisation_attempts, max_acceptable_value, seed, context, verbose)
+  function diver(func, &
+                 lowerbounds, &
+                 upperbounds, &
+                 path, &
+                 nDerived, &
+                 bestFitParams, &
+                 bestFitDerived, &
+                 discrete, &
+                 partitionDiscrete, &
+                 maxciv, &
+                 maxgen, &
+                 NP, &
+                 F, &
+                 Cr, &
+                 lambda, &
+                 current, &
+                 expon, &
+                 bndry, &
+                 jDE, &
+                 lambdajDE, &
+                 convthresh, &
+                 convsteps, &
+                 removeDuplicates, &
+                 doBayesian, &
+                 prior, &
+                 maxNodePop, &
+                 Ztolerance, &
+                 savecount, &
+                 resume, &
+                 disableIO, &
+                 outputRaw, &
+                 outputSam, &
+                 init_population_strategy, &
+                 discard_unfit_points, &
+                 max_initialisation_attempts, &
+                 max_acceptable_value, &
+                 seed, &
+                 context, &
+                 verbose)
 
     use iso_c_binding, only: c_ptr
 
+    real(dp)                         :: diver                   !result (minimum value obtained from func)
     procedure(MinusLogLikeFunc)      :: func                    !the likelihood function to be minimised -- assumed to be -ln(likelihood)
     real(dp), dimension(:), intent(in) :: lowerbounds, upperbounds !boundaries of parameter space
     character(len=*), intent(in)     :: path                    !path to save samples, resume files, etc
```

Comment:
Oh, I just realized, if `disableIO == .true.`, then there's not much point in giving the function a path. Can we make this an optional parameter as well?

Reply:
yes, probably - I'll just have to poke about a bit and ensure that the `path` variable is not still passed around when `disableIO == .true.`

Reply:
Good point. I suppose in that case we could just initialize it to `'./'`

Reply:
OK, I've made it optional now and propagated the logic through to where it's used, then thrown errors in cases where it is needed but hasn't been defined.